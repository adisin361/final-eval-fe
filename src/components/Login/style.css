@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@300;500;700&display=swap');

.login-page {
  display: flex;
  height: 100vh;
  font-family: Montserrat;
}

.image-section {
  display: flex;
  flex-direction: column;
  width: 60vw;
  padding-top: 10vh;
  background: rgb(236, 234, 250);
}

.image-section p {
  margin-left: 10vw;
  font-family: Montserrat;
  font-size: 2.5rem;
  font-weight: 700;
  color: rgb(67, 67, 67);
  margin-bottom: 15px;
}

.image-section img {
  width: 50%;
  align-self: center;
}

.login-section {
  width: 40vw;
  display: flex;
  flex-direction: column;
  align-items: center;
  background: rgb(42, 42, 42);
}

.login-section>p {
  font-size: 2rem;
  font-weight: 600;
  margin-top: 15vh;
  margin-bottom: 15vh;
}

.login-section p {
  color: white;
}

.login-form {
  font-weight: 300;
  font-size: 0.9rem;
}

.login-form input {
  width: 16vw;
  height: 4vh;
  border-radius: 6px;
  border: none;
  padding-left: 10px;
  padding-right: 10px;
  outline: none;
  margin-bottom: 14px;
  margin-top: 8px;
}

.login-form button {
  width: 16vw;
  height: 4vh;
  border-radius: 6px;
  border: none;
  margin-top: 22px;
  margin-bottom: 15px;
  background: rgb(135, 36, 168);
  background: linear-gradient(90deg, rgba(135, 36, 168, 1) 0%, rgba(0, 79, 217, 1) 100%);
  color: white;
  font-family: Montserrat;
  font-weight: 500;
  cursor: pointer;
}

#forgot-pwd {
  text-align: center;
  text-decoration: underline;
  cursor: pointer;
}