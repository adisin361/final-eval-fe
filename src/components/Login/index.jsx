/* eslint-disable no-unused-vars */
import './style.css';
import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import bgImg from '../../assets/undraw-upload-re-pasx_2023-03-09/undraw-upload-re-pasx.png';
import { makeAuthRequest } from '../../utils/makeRequest';
import { LOGIN_URL } from '../../constants/apiEndPoints';
const Login = () => {

  const navigate = useNavigate();
  const [user, setUser] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState(null);

  const handleSubmit = async () => {
    try {

      const loginData = await makeAuthRequest(LOGIN_URL, {
        data: {
          username: user,
          password,
        },
      });
      localStorage.setItem('jwt_token', loginData.data.token);
      navigate('/dashboard');
    } catch (error) {
      setError(error);
      setTimeout(() => {
        setError(null);
      }, 4000);
    }
  };
  return (
    <>
      <div className='login-page'>
        <div className='image-section'>
          <p>Design APIs fast,</p>
          <p>Manage content easily</p>
          <img src={bgImg} />
        </div>

        <div className='login-section'>
          <p>Login to your CMS+ account</p>
          <div className='login-form'>
            <p>Email</p>
            <input type="email"
              name="username"
              value={user}
              onChange={(event) => setUser(event.target.value)} />

            <p>Password</p>
            <input type="password"
              name="password"
              value={password}
              onChange={(event) => setPassword(event.target.value)}
            />
            <br />
            <button type="submit"
              disabled={user === '' || password === ''}
              onClick={handleSubmit} data-testid='register-link'>Login</button>
            <p id='forgot-pwd'>Forgot password?</p>
          </div>
        </div>
      </div>
    </>
  );
};

export default Login;