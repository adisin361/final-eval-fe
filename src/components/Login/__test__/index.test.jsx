import React from 'react';
import Login from '..';
import { render } from '@testing-library/react';

const mockedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  useNavigate: () => mockedNavigate,
}));

describe('Login Page', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Login />);
    expect(asFragment()).toMatchSnapshot();
  });
});
