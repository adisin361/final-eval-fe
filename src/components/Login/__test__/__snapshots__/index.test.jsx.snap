// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should render correctly 1`] = `
<DocumentFragment>
  <div
    class="login-page"
  >
    <div
      class="image-section"
    >
      <p>
        Design APIs fast,
      </p>
      <p>
        Manage content easily
      </p>
      <img
        src="undraw-upload-re-pasx.png"
      />
    </div>
    <div
      class="login-section"
    >
      <p>
        Login to your CMS+ account
      </p>
      <div
        class="login-form"
      >
        <p>
          Email
        </p>
        <input
          name="username"
          type="email"
          value=""
        />
        <p>
          Password
        </p>
        <input
          name="password"
          type="password"
          value=""
        />
        <br />
        <button
          data-testid="register-link"
          disabled=""
          type="submit"
        >
          Login
        </button>
        <p
          id="forgot-pwd"
        >
          Forgot password?
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
